File: backend/app/entities/classification_rule.py
from __future__ import annotations

from sqlalchemy import Integer, Text
from sqlalchemy.orm import Mapped, mapped_column

from backend.app.core.database import Base


class ClassificationRule(Base):
    __tablename__ = "classification_rules"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    rule_group: Mapped[str] = mapped_column(Text, nullable=False)
    target_type: Mapped[str] = mapped_column(Text, nullable=False)
    rule_name: Mapped[str] = mapped_column(Text, nullable=False)
    keywords: Mapped[str] = mapped_column(Text, nullable=False)
    output_field: Mapped[str] = mapped_column(Text, nullable=False)
    output_value: Mapped[str] = mapped_column(Text, nullable=False)
    score_delta: Mapped[int] = mapped_column(Integer, nullable=False, default=0)
    priority: Mapped[int] = mapped_column(Integer, nullable=False, default=100)
    is_active: Mapped[int] = mapped_column(Integer, nullable=False, default=1)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    created_at: Mapped[str] = mapped_column(Text, nullable=False)
    updated_at: Mapped[str] = mapped_column(Text, nullable=False)

File: frontend/src/types/classificationRule.ts
export type ClassificationRule = {
  id: number;
  rule_group: string;
  target_type: string;
  rule_name: string;
  keywords: string;
  output_field: string;
  output_value: string;
  score_delta: number;
  priority: number;
  is_active: boolean;
  description: string | null;
  created_at: string;
  updated_at: string;
};

export type ClassificationRuleListParams = {
  target_type?: string;
  rule_group?: string;
  is_active?: boolean;
  keyword?: string;
  limit?: number;
  offset?: number;
};

export type ClassificationRuleCreatePayload = {
  rule_group: string;
  target_type: string;
  rule_name: string;
  keywords: string;
  output_field: string;
  output_value: string;
  score_delta?: number;
  priority?: number;
  is_active?: boolean;
  description?: string | null;
};

export type ClassificationRuleUpdatePayload = Partial<ClassificationRuleCreatePayload>;
